packages/scene-composer/src/components/three-fiber/AnimationComponent/AnimationComponent.tsx

```diff
@@ -0,0 +1,81 @@
+// eslint-disable-next-line import/default
+import React, { Fragment, useEffect } from 'react';
+import * as THREE from 'three';
+import { useFrame } from '@react-three/fiber';
+
+import { useSceneComposerId } from '../../../common/sceneComposerIdContext';
+import { ISceneNodeInternal, useEditorState, useViewOptionState, IAnimationComponentInternal } from '../../../store';
+import { IAnimationComponent } from '../../../interfaces';
+
+const toggleIsAnimationPaused = (
+  object: THREE.Object3D,
+  AnimationMixer: THREE.AnimationMixer,
+  isPaused: boolean,
+  AnimationList: string[],
+) => {
+  try {
+    AnimationList.forEach(function (clip) {
+      const animation = THREE.AnimationClip.findByName(object.animations, clip);
+      AnimationMixer.clipAction(animation).paused = isPaused;
+    });
+  } catch (error) {
+    // console.error(error);
+  }
+};
+
+const list: any[] = [];
+interface AnimationComponentProps {
+  component: IAnimationComponent;
+  node: ISceneNodeInternal;
+}
+
+const AnimationComponent: React.FC<AnimationComponentProps> = ({ component, node }) => {
+  const sceneComposerId = useSceneComposerId();
+  const isGlobalAnimationPaused = !useViewOptionState(sceneComposerId).componentVisibilities['Animation'];
+  let obj;
+  const { getObject3DBySceneNodeRef } = useEditorState(sceneComposerId);
+  const object = getObject3DBySceneNodeRef(node.ref);
+  const currentAnimations: any[] = [];
+  if (object) {
+    obj = object.getObjectByName('Scene');
+    currentAnimations.push(component.currentAnimations);
+  }
+
+  let AnimationMixer: THREE.AnimationMixer;
+  useEffect(() => {
+    if (obj) {
+      list.push({ component, obj });
+      AnimationMixer = new THREE.AnimationMixer(obj);
+    }
+    if (obj?.animations) {
+      currentAnimations?.forEach(function (clipGroup) {
+        clipGroup?.forEach(function (clip) {
+          const animation = THREE.AnimationClip.findByName(obj, clip);
+          AnimationMixer.clipAction(animation).play();
+        });
+        toggleIsAnimationPaused(obj, AnimationMixer, isGlobalAnimationPaused, clipGroup);
+      });
+    }
+  }, [object, obj, isGlobalAnimationPaused, currentAnimations]);
+
+  useFrame((state, delta) => {
+    if (AnimationMixer) {
+      AnimationMixer.update(delta);
+    }
+  });
+
+  return <Fragment />;
+};
+export function getClip(component: IAnimationComponentInternal): THREE.Object3D | undefined {
```

Comment:
This isn't a getClip function. This is a get object3D function that gets consumed elsewhere for the animation list... which leads to the variable name being misleading (it's animationList but it's actually an object3D). We can just return the obj.animations instead and update the return type if we want to get all the clips.